packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js

```diff
@@ -0,0 +1,182 @@
+import find from 'lodash/find';
+import filter from 'lodash/filter';
+import get from 'lodash/get';
+import map from 'lodash/map';
+import sortBy from 'lodash/sortBy';
+
+import { NAME_PATTERN } from './add.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($translate, $q, CucCloudMessage, DatabaseService, Poller) {
+    this.$translate = $translate;
+    this.$q = $q;
+    this.CucCloudMessage = CucCloudMessage;
+    this.DatabaseService = DatabaseService;
+    this.Poller = Poller;
+    this.NAME_PATTERN = NAME_PATTERN;
+  }
+
+  $onInit() {
+    this.loadMessages();
+    this.model = {
+      engine: find(this.engines, 'isDefault'),
+      plan: null,
+      region: null,
+      flavor: null,
+      name: '',
+      privateNetwork: null,
+      subnet: null,
+      usePrivateNetwork: false,
+    };
+    this.orderData = null;
+    this.defaultPrivateNetwork = {
+      id: '',
+      name: this.$translate.instant('pci_database_common_none'),
+    };
+    this.defaultSubnet = {
+      id: '',
+      name: this.$translate.instant('pci_database_common_none'),
+    };
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(
+      'pci.projects.project.storages.databases.add',
+    );
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      'pci.projects.project.storages.databases.add',
+      { onMessage: () => this.refreshMessages() },
+    );
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  setDefaultPlan() {
+    this.model.plan = this.model.engine.selectedVersion.plans.includes(
+      this.model.plan,
+    )
+      ? this.model.plan
+      : this.model.engine.selectedVersion.getDefaultPlan();
```

Comment:
**sugestion**:
Why not passing `this.model.plan` as a parameter like `selectedPlan` ? It would be more readable to test the value, and set it to `this.model`if it satisfies your test